BSP/STM32F413/Src/BSP_SPI.C

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "BSP_SPI.h"
+#include "stm32f4xx.h"
+#include "os.h"
+
+
+#define SPI_PORT SPI3
+
+static OS_SEM SPI_Update_Sem4;
+
+static void spi_post(void) {
+	OS_ERR err;
+	OSSemPost(&SPI_Update_Sem4, OS_OPT_POST_1, &err);
+	// TODO: error handling
+}
+
+static void spi_pend(void) {
+	OS_ERR err;
+	CPU_TS ts;
+	OSSemPend(&SPI_Update_Sem4, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
+	// TODO: error handling
+}
+
+// Use this inline function to wait until SPI communication is complete
+static inline void SPI_Wait(){
+	spi_pend();
+}
+
+/** SPI_WriteRead
+ * @brief   Sends and receives a byte of data on the SPI line.
+ * @param   txData single byte that will be sent to the device.
+ * @return  rxData single byte that was read from the device.
+ */
+static uint8_t SPI_WriteRead(uint8_t txData){
+
+	SPI_TypeDef *bus = SPI_PORT;
+	
+	SPI_Wait(bus);
```

Comment:
Is this supposed to take a parameter? From the function definition above it seems like it doesn't